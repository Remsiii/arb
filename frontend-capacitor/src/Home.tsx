import React, { FC } from 'react';
import MenuCard from './MenuCard'; // Pfad anpassen, falls nötig
import styles from './MenuList.module.css'; // Pfad zur CSS-Moduldatei anpassen
import menuData from './menuData.json';

// Typdefinition für ein Menüelement
interface MenuItem {
  type: string;
  title: string;
  description: string;
  price: number;
}

// Die Menüdaten könnten dynamisch importiert werden oder als Prop übergeben werden

const MenuList: FC = () => {
  return (
    <div className={styles.menuContainer}>
      {menuData.map((item: MenuItem, index: number) => (
        <div key={index} className={styles.menuItem}>
          <h2>{item.type}</h2>
          <MenuCard 
            title={item.title} 
            description={item.description} 
            price={item.price} 
            // Hier müsste eine Funktion übergeben werden, wenn "addToCart" verwendet wird.
          />
        </div>
      ))}
      <p>page</p>
    </div>
  );
};

export default MenuList;
