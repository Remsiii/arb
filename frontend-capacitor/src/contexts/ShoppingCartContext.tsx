// ShoppingCartContext.tsx

import React, { createContext, useState, useContext } from 'react';

type CartItem = {
  title: string;
  description: string;
  price: number;
  type: string;
  quantity: number;
};

type ShoppingCartProviderProps = {
    children: React.ReactNode;
  };

type ShoppingCartContextType = {
  items: CartItem[];
  setItems: React.Dispatch<React.SetStateAction<CartItem[]>>;
};

const ShoppingCartContext = createContext<ShoppingCartContextType | undefined>(undefined);

export const ShoppingCartProvider: React.FC<ShoppingCartProviderProps> = ({ children }) => {
  const [items, setItems] = useState<CartItem[]>([]);

  return (
    <ShoppingCartContext.Provider value={{ items, setItems }}>
      {children}
    </ShoppingCartContext.Provider>
  );
};

export const useShoppingCart = (): ShoppingCartContextType => {
  const context = useContext(ShoppingCartContext);
  if (!context) {
    throw new Error('useShoppingCart muss innerhalb eines ShoppingCartProvider verwendet werden.');
  }
  return context;
};


