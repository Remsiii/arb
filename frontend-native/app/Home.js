import React, { useState, useRef } from 'react';
import { ScrollView, SafeAreaView, StyleSheet, TouchableOpacity, Text, View } from 'react-native';
import MenuCard from '../components/RecipeCard';
import menuData from './menuData.json';
import Arrows from '../components/Arrows';

const Home = ({ navigation }) => {
  const [selectedType, setSelectedType] = useState('Vorspeise');
  const [cartItems, setCartItems] = useState([]);
  const [cartCount, setCartCount] = useState(0);

  const uniqueTypes = [...new Set(menuData.map(item => item.type))];

  const scrollViewRef = useRef(null);
  
  const openShoppingCart = () => {
    navigation.navigate('ShoppingCart', { cartItems });
  };

  const handleArrowClick = (direction) => {
    const currentIndex = uniqueTypes.indexOf(selectedType);
    const itemWidth = 100;  // Dies sollte die ungefÃ¤hre Breite deiner navItems sein. Passe dies entsprechend an.

    if (direction === 'left' && currentIndex > 0) {
      setSelectedType(uniqueTypes[currentIndex - 1]);
      scrollViewRef.current.scrollTo({ x: itemWidth * (currentIndex - 1), animated: true });
    } else if (direction === 'right' && currentIndex < uniqueTypes.length - 1) {
      setSelectedType(uniqueTypes[currentIndex + 1]);
      scrollViewRef.current.scrollTo({ x: itemWidth * currentIndex, animated: true });
    }
  };

  const addToCart = (item) => {
    setCartItems([...cartItems, item]);
  };

  return (
    <View style={{ flex: 1 }}>
      <SafeAreaView style={styles.container}>
        <View style={styles.navbarWrapper}>
          <Arrows handleArrowClick={handleArrowClick} direction="left" />


          <ScrollView horizontal ref={scrollViewRef} style={styles.navbar}>
            {uniqueTypes.map((type) => (
              <TouchableOpacity key={type} style={[styles.navItem, selectedType === type && styles.navItemSelected]} onPress={() => setSelectedType(type)}>
                <Text style={selectedType === type ? styles.navTextSelected : styles.navText}>{type}</Text>
              </TouchableOpacity>
            ))}
          </ScrollView>

          <Arrows handleArrowClick={handleArrowClick} direction="right" />

        </View>

        <ScrollView style={styles.contentScrollView}>
          {menuData.filter(item => item.type === selectedType).map((item, index) => (
            <MenuCard
              key={index}
              title={item.title}
              description={item.description}
              price={item.price}
              setCartCount={setCartCount}
              addToCart={() => addToCart(item)}  // Funktion als Prop weitergegeben
            />
          ))}
        </ScrollView>
        {cartCount > 0 && (
          <TouchableOpacity style={styles.cartButton} onPress={openShoppingCart}>
            <Text style={styles.cartIcon}>ðŸ›’</Text>
            {cartCount > 0 && <Text style={styles.cartCount}>{cartCount}</Text>}
          </TouchableOpacity>
        )}
      </SafeAreaView>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#f4f4f4',
    padding: 10,
  },
  navbarWrapper: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    marginBottom: 25,
    borderBottomWidth: 1,
    borderBottomColor: '#ccc',
    paddingVertical: 5,
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    zIndex: 1,
    backgroundColor: '#ffffff',
  },
  arrowText: {
    fontSize: 18,
    marginHorizontal: 10,
    color: '#000',
  },
  navbar: {
    flex: 1,
    flexDirection: 'row',
    marginHorizontal: 10,
  },
  navItem: {
    marginHorizontal: 5,
    paddingVertical: 15,
    paddingHorizontal: 15,
    borderRadius: 10,
    backgroundColor: '#f5f5f5'
  },
  navItemSelected: {
    backgroundColor: '#e0e0e0'
  },
  navText: {
    fontSize: 16,
    color: '#333'
  },
  navTextSelected: {
    fontSize: 16,
    fontWeight: 'bold',
    color: '#000'
  },
  contentScrollView: {
    marginTop: 60,
  },
  cartIcon: {
    position: 'absolute',
    bottom: 50,
    right: 30,
    height: 50,
    width: 50,
    borderRadius: 30,
    backgroundColor: '#C5E1A5',
    alignItems: 'center',
    justifyContent: 'center'
  },

  cartButton: {
    position: 'absolute',
    bottom: 20,
    right: 20,
    backgroundColor: '#A3C492',
    borderRadius: 30,
    width: 60,
    height: 60,
    alignItems: 'center',
    justifyContent: 'center',
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 3 },
    shadowOpacity: 0.3,
    shadowRadius: 4,
    elevation: 5
  },
  cartIcon: {
    fontSize: 24,
  },
  cartCount: {
    position: 'absolute',
    top: 5,
    right: 5,
    backgroundColor: 'red',   // Roten Hintergrund fÃ¼r bessere Sichtbarkeit
    borderRadius: 8,  // KreisfÃ¶rmigen Hintergrund
    paddingHorizontal: 5,
    fontSize: 12,
    color: '#FFF',
    fontWeight: 'bold'
  }
});

export default Home;
