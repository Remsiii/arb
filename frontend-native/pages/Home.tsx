import React, { useState, useRef, FC } from 'react';
import { ScrollView, SafeAreaView, StyleSheet, TouchableOpacity, Text, View } from 'react-native';
import MenuCard from '../components/RecipeCard';
import menuData from '../app/menuData.json';
import Arrows from '../components/Arrows';

interface CartItem {
  title: string;
  description: string;
  price: number;
  type: string;
  quantity?: number;
}

interface HomeProps {
  navigation: {
    navigate: (route: string, params: { cartItems: CartItem[] }) => void;
  };
}

const Home: FC<HomeProps> = ({ navigation }) => {
  const [selectedType, setSelectedType] = useState<string>('Vorspeise');
  const [cartItems, setCartItems] = useState<CartItem[]>([]);
  const [cartCount, setCartCount] = useState<number>(0);

  const uniqueTypes = [...new Set(menuData.map(item => item.type))];

  const scrollViewRef = useRef<ScrollView>(null);

  const openShoppingCart = () => {
    navigation.navigate('ShoppingCart', { cartItems });
  };

  const handleArrowClick = (direction: string) => {
    const currentIndex = uniqueTypes.indexOf(selectedType);
    const itemWidth = 100;

    if (direction === 'left' && currentIndex > 0) {
      setSelectedType(uniqueTypes[currentIndex - 1]);
      scrollViewRef.current?.scrollTo({ x: itemWidth * (currentIndex - 1), animated: true });
    } else if (direction === 'right' && currentIndex < uniqueTypes.length - 1) {
      setSelectedType(uniqueTypes[currentIndex + 1]);
      scrollViewRef.current?.scrollTo({ x: itemWidth * currentIndex, animated: true });
    }
  };

  const addToCart = (item: CartItem) => {
    let updatedCartItems = [...cartItems];
    const existingItemIndex = updatedCartItems.findIndex(cartItem => cartItem.title === item.title);

    if (existingItemIndex !== -1) {
        updatedCartItems[existingItemIndex].quantity = (updatedCartItems[existingItemIndex].quantity || 1) + 1;
    } else {
        item.quantity = 1;
        updatedCartItems = [...updatedCartItems, item];
    }

    // Anzahl der Artikel im Warenkorb aktualisieren
    const newCartCount = updatedCartItems.reduce((acc, currentItem) => acc + (currentItem.quantity || 1), 0);
    setCartItems(updatedCartItems);
    setCartCount(newCartCount);
};


  return (
    <View style={{ flex: 1 }}>
      <SafeAreaView style={styles.container}>
        <View style={styles.navbarWrapper}>
          <Arrows handleArrowClick={handleArrowClick} direction="left" />
          <ScrollView horizontal ref={scrollViewRef} style={styles.navbar}>
            {uniqueTypes.map((type) => (
              <TouchableOpacity key={type} style={[styles.navItem, selectedType === type && styles.navItemSelected]} onPress={() => setSelectedType(type)}>
                <Text style={selectedType === type ? styles.navTextSelected : styles.navText}>{type}</Text>
              </TouchableOpacity>
            ))}
          </ScrollView>
          <Arrows handleArrowClick={handleArrowClick} direction="right" />
        </View>
        <ScrollView style={styles.contentScrollView}>
          {menuData.filter(item => item.type === selectedType).map((item, index) => (
            <MenuCard
              key={index}
              title={item.title}
              description={item.description}
              price={item.price}
              addToCart={() => addToCart(item)}
            />
          ))}
        </ScrollView>
        {cartCount > 0 && (
          <TouchableOpacity style={styles.cartButton} onPress={openShoppingCart}>
            <Text style={styles.cartIcon}>ðŸ›’</Text>
            {cartCount > 0 && <Text style={styles.cartCount}>{cartCount}</Text>}
          </TouchableOpacity>
        )}
      </SafeAreaView>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#f4f4f4',
    padding: 10,
  },
  navbarWrapper: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    marginBottom: 25,
    borderBottomWidth: 1,
    borderBottomColor: '#ccc',
    paddingVertical: 5,
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    zIndex: 1,
    backgroundColor: '#ffffff',
  },
  navbar: {
    flex: 1,
    flexDirection: 'row',
    marginHorizontal: 10,
  },
  navItem: {
    marginHorizontal: 5,
    paddingVertical: 15,
    paddingHorizontal: 15,
    borderRadius: 10,
    backgroundColor: '#f5f5f5'
  },
  navItemSelected: {
    backgroundColor: '#e0e0e0'
  },
  navText: {
    fontSize: 16,
    color: '#333'
  },
  navTextSelected: {
    fontSize: 16,
    fontWeight: 'bold',
    color: '#000'
  },
  contentScrollView: {
    marginTop: 60,
  },
  cartButton: {
    position: 'absolute',
    bottom: 20,
    right: 20,
    backgroundColor: '#A3C492',
    borderRadius: 30,
    width: 60,
    height: 60,
    alignItems: 'center',
    justifyContent: 'center',
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 3 },
    shadowOpacity: 0.3,
    shadowRadius: 4,
    elevation: 5
  },
  cartIcon: {
    fontSize: 24,
  },
  cartCount: {
    position: 'absolute',
    top: 5,
    right: 5,
    backgroundColor: 'red',
    borderRadius: 8,
    paddingHorizontal: 5,
    fontSize: 12,
    color: '#FFF',
    fontWeight: 'bold'
  }
});

export default Home;
