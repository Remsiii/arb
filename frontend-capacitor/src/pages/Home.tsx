import React, { useState, useRef, useMemo } from 'react';
import { useNavigate } from 'react-router-dom';
import MenuCard from '../components/RecipeCard';
import menuData from '../menuData.json';
import Arrows from '../components/Arrows';
import { useShoppingCart } from '../contexts/ShoppingCartContext';
import { CartItem } from '../types/CartItems';

const Home = () => {
  const [selectedType, setSelectedType] = useState('Vorspeise');
  const { items, setItems } = useShoppingCart();
  const navigate = useNavigate();
  const uniqueTypes = Array.from(new Set(menuData.map(item => item.type))); 
  const scrollViewRef = useRef(null);

  const openShoppingCart = () => {
    navigate('/shopping-cart');
  };

  const handleArrowClick = (direction: 'left' | 'right') => {
    // Logik fÃ¼r das Klicken der Pfeile...
  };

  const addToCart = (item: Omit<CartItem, 'quantity'>) => {
    let updatedCartItems = [...items];
    const existingItemIndex = updatedCartItems.findIndex(cartItem => cartItem.title === item.title);
  
    if (existingItemIndex !== -1) {
      // Wenn der Artikel bereits im Warenkorb ist, erhÃ¶he die Menge
      updatedCartItems[existingItemIndex].quantity += 1;
    } else {
      // Wenn der Artikel neu ist, fÃ¼ge ihn mit Menge 1 hinzu
      updatedCartItems.push({ ...item, quantity: 1 });
    }
  
    setItems(updatedCartItems);
  };
  

  const cartCount = useMemo(() => {
    return items.reduce((acc, item) => acc + item.quantity, 0);
  }, [items]);

  return (
    <div className='safe-area'>
      <div className='flex flex-col'>
        {/* Arrows component */}
        {/* <Arrows handleArrowClick={handleArrowClick} direction={selectedType} /> */}
        {/* Content scroll view */}
        <div className='content-scroll-view overflow-x-scroll'>
          {uniqueTypes.map((type) => (
            menuData.filter(item => item.type === type).map((item, index) => (
              <MenuCard key={index} {...item} addToCart={() => addToCart(item)} />
            ))
          ))}
          <div className='text-red leading-7'>asd</div>
        </div>
        {/* Cart button */}
        <button 
          className='fixed bottom-4 right-4 bg-blue-500 text-white p-3 rounded-full shadow-lg flex items-center justify-center'
          onClick={openShoppingCart}
        >
          <span>ðŸ›’</span>
          <span className='ml-2'>{cartCount}</span>
        </button>
      </div>
    </div>
  );
};

export default Home;
