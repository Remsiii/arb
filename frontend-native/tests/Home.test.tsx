
import React from 'react';
import { render } from '@testing-library/react-native';
import Home from '../pages/Home';
import { ShoppingCartProvider } from '../contexts/ShoppingCartContext';

// Mocking the navigation hook
jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({
      navigate: jest.fn(),
    }),
  };
});

describe('<Home />', () => {
  it('renders correctly', () => {
    const tree = render(
      <ShoppingCartProvider>
        <Home navigation={{
          navigate: function (route: string, params: { cartItems: { title: string; description: string; price: number; type: string; quantity: number; }[]; }): void {
            throw new Error('Function not implemented.');
          }
        }} />
      </ShoppingCartProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

export {};
