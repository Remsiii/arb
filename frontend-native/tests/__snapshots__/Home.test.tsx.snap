// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    style={
      {
        "backgroundColor": "#f4f4f4",
        "flex": 1,
        "padding": 10,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderBottomColor": "#ccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "left": 0,
          "marginBottom": 25,
          "paddingVertical": 5,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 1,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            {
              "color": "#000",
              "fontSize": 18,
              "marginHorizontal": 10,
            }
          }
        >
          &lt;
        </Text>
      </View>
      <RCTScrollView
        horizontal={true}
        style={
          {
            "flex": 1,
            "flexDirection": "row",
            "marginHorizontal": 10,
          }
        }
      >
        <View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#e0e0e0",
                "borderRadius": 10,
                "marginHorizontal": 5,
                "opacity": 1,
                "paddingHorizontal": 15,
                "paddingVertical": 15,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#000",
                  "fontSize": 16,
                  "fontWeight": "bold",
                }
              }
            >
              Vorspeise
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#f5f5f5",
                "borderRadius": 10,
                "marginHorizontal": 5,
                "opacity": 1,
                "paddingHorizontal": 15,
                "paddingVertical": 15,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#333",
                  "fontSize": 16,
                }
              }
            >
              Hauptgericht
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#f5f5f5",
                "borderRadius": 10,
                "marginHorizontal": 5,
                "opacity": 1,
                "paddingHorizontal": 15,
                "paddingVertical": 15,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#333",
                  "fontSize": 16,
                }
              }
            >
              Nachspeise
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "backgroundColor": "#f5f5f5",
                "borderRadius": 10,
                "marginHorizontal": 5,
                "opacity": 1,
                "paddingHorizontal": 15,
                "paddingVertical": 15,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#333",
                  "fontSize": 16,
                }
              }
            >
              Getr√§nke
            </Text>
          </View>
        </View>
      </RCTScrollView>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            {
              "color": "#000",
              "fontSize": 18,
              "marginHorizontal": 10,
            }
          }
        >
          &gt;
        </Text>
      </View>
    </View>
    <RCTScrollView
      style={
        {
          "marginTop": 60,
        }
      }
    >
      <View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "margin": 10,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": "#FFF",
                "borderRadius": 8,
                "elevation": 3,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "padding": 15,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 0.2,
                "shadowRadius": 2,
              }
            }
          >
            <View
              style={
                {
                  "flex": 3,
                }
              }
            >
              <Text
                style={
                  {
                    "fontSize": 18,
                    "fontWeight": "bold",
                  }
                }
              >
                Mancacasi Mami
              </Text>
              <Text
                style={
                  {
                    "color": "#888",
                    "fontSize": 14,
                    "marginTop": 8,
                  }
                }
              >
                Calamari prƒÉji»õi √Æntr-un mix de fƒÉinƒÉ aromatizatƒÉ cu fenicul »ôi oregano, acompania»õi de un sos u»ôor picant de maionezƒÉ »ôi harissa. Totul este pus √Æn valoare de un mix proaspƒÉt de salatƒÉ, cu dressing delicios de lƒÉm√¢ie »ôi miere.
              </Text>
            </View>
            <View
              style={
                {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                }
              }
            >
              <Text
                style={
                  {
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginRight": 10,
                  }
                }
              >
                10
                ‚Ç¨
              </Text>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#C5E1A5",
                    "borderRadius": 15,
                    "height": 30,
                    "justifyContent": "center",
                    "opacity": 1,
                    "width": 30,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#4E7C5B",
                      "fontSize": 25,
                    }
                  }
                >
                  +
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "margin": 10,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": "#FFF",
                "borderRadius": 8,
                "elevation": 3,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "padding": 15,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 0.2,
                "shadowRadius": 2,
              }
            }
          >
            <View
              style={
                {
                  "flex": 3,
                }
              }
            >
              <Text
                style={
                  {
                    "fontSize": 18,
                    "fontWeight": "bold",
                  }
                }
              >
                Lol Imagone
              </Text>
              <Text
                style={
                  {
                    "color": "#888",
                    "fontSize": 14,
                    "marginTop": 8,
                  }
                }
              >
                Calamari prƒÉji»õi √Æntr-un mix de fƒÉinƒÉ aromatizatƒÉ cu fenicul »ôi oregano, acompania»õi de un sos u»ôor picant de maionezƒÉ »ôi harissa. Totul este pus √Æn valoare de un mix proaspƒÉt de salatƒÉ, cu dressing delicios de lƒÉm√¢ie »ôi miere.
              </Text>
            </View>
            <View
              style={
                {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                }
              }
            >
              <Text
                style={
                  {
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginRight": 10,
                  }
                }
              >
                23.67
                ‚Ç¨
              </Text>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#C5E1A5",
                    "borderRadius": 15,
                    "height": 30,
                    "justifyContent": "center",
                    "opacity": 1,
                    "width": 30,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#4E7C5B",
                      "fontSize": 25,
                    }
                  }
                >
                  +
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "margin": 10,
              "opacity": 1,
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": "#FFF",
                "borderRadius": 8,
                "elevation": 3,
                "flexDirection": "row",
                "justifyContent": "space-between",
                "padding": 15,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 1,
                  "width": 0,
                },
                "shadowOpacity": 0.2,
                "shadowRadius": 2,
              }
            }
          >
            <View
              style={
                {
                  "flex": 3,
                }
              }
            >
              <Text
                style={
                  {
                    "fontSize": 18,
                    "fontWeight": "bold",
                  }
                }
              >
                CALAMARI GANG
              </Text>
              <Text
                style={
                  {
                    "color": "#888",
                    "fontSize": 14,
                    "marginTop": 8,
                  }
                }
              >
                Calamari prƒÉji»õi √Æntr-un mix de fƒÉinƒÉ aromatizatƒÉ cu fenicul »ôi oregano, acompania»õi de un sos u»ôor picant de maionezƒÉ »ôi harissa. Totul este pus √Æn valoare de un mix proaspƒÉt de salatƒÉ, cu dressing delicios de lƒÉm√¢ie »ôi miere.
              </Text>
            </View>
            <View
              style={
                {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                }
              }
            >
              <Text
                style={
                  {
                    "fontSize": 16,
                    "fontWeight": "bold",
                    "marginRight": 10,
                  }
                }
              >
                100
                ‚Ç¨
              </Text>
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "#C5E1A5",
                    "borderRadius": 15,
                    "height": 30,
                    "justifyContent": "center",
                    "opacity": 1,
                    "width": 30,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#4E7C5B",
                      "fontSize": 25,
                    }
                  }
                >
                  +
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>
</View>
`;
